src/SIL/Parser.hs

```diff
@@ -273,7 +258,36 @@ parseApplied = do
   fargs <- L.lineFold scn $ \sc' ->
     parseSingleExpr `sepBy` try sc'
   case fargs of
-    (f:args) -> pure $ foldl TApp f args
+    (f:args) -> do
+      case f of
+        TVar (Right "left") -> case args of
+          [t] -> pure . TLeft $ t
+          [] -> fail "This should be imposible. I'm being called fro parseApplied."
+          _ -> fail "Failed to parse left. Too many arguments applied to left."
```

Comment:
We actually want to support more arguments. Consider `left (\x -> {x,0}, 0) 0`. Please add a test case for this.

Reply:
apart from left and right, should any AST function accept any amount of applications?

Reply:
`pair (\x y z -> x) a b c d`
maybe that should evaluate to `{a, d}`

but it seems weird. I would personally always use parenthesis to delimit.

Reply:
no, that example shouldn't type check, because you're applying `c` to a pair, not a function